aspnetcore/blazor/components/index.md

```diff
@@ -737,6 +737,28 @@ Single-line attribute lists are also supported:
 public string? Title { get; set; }
 ```
 
+:::moniker-end
+
+:::moniker range=">= aspnetcore-7.0"
+
+Don't use the [`required` modifier](/dotnet/csharp/language-reference/keywords/required) on component parameter properties seeking to require an assigned value. Component parameters aren't assigned values on component instantiation. Instead, a <xref:Microsoft.AspNetCore.Components.ParameterView> object is passed to the component's <xref:Microsoft.AspNetCore.Components.ComponentBase.SetParametersAsync%2A> method to assign parameter property values. The `required` modifier has no bearing on whether a parameter has an entry in the <xref:Microsoft.AspNetCore.Components.ParameterView>. Even for parameters that do have <xref:Microsoft.AspNetCore.Components.ParameterView> entries, their properties don't receive values until after the time when the `required` modifier is supposed to guarantee a value assignment. Likewise, don't use the [`init` accessor](/dotnet/csharp/language-reference/keywords/init) on component parameter properties because setting component parameter values with <xref:Microsoft.AspNetCore.Components.ParameterView.SetParameterProperties%2A?displayProperty=nameWithType> uses reflection, which bypasses the init-only setter restriction. For the preceding reasons, only use the [`[EditorRequired]` attribute](xref:Microsoft.AspNetCore.Components.EditorRequiredAttribute) to specify a required component parameter.
```

Comment:
I think we can probably shorten this down. There's a lot of technical detail here that a most customers probably won't care too much about 🙂 

Maybe something like:

```suggestion
Don't use the [`required` modifier](/dotnet/csharp/language-reference/keywords/required) or [`init` accessor](/dotnet/csharp/language-reference/keywords/init) on component parameter properties. Components are usually instantiated and assigned parameter values using reflection, which bypasses the guarantees that `init` and `required` are designed to make. Instead, use the [`[EditorRequired]` attribute](xref:Microsoft.AspNetCore.Components.EditorRequiredAttribute) to specify a required component parameter.
```